### PR TITLE
docs: add the sheets quickstart

### PR DESCRIPTION
**Pull request by JustinBeckwith**
We're seeing a bunch of reports of old docs from codesite.  This is the first step to fixing #999.   We will need to follow up with tests, and then submit a CL to embed the code snippet.

### CODE REVIEW

File: samples/sheets/quickstart.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    auth: auth,
+    spreadsheetId: '1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms',
+    range: 'Class Data!A2:E',
+  }, (err, res) => {
+    if (err) {
+      console.error('The API returned an error.');
+      throw err;
+    }
+    const rows = res.data.values;
+    if (rows.length == 0) {
```
Comment (alexander-fenster) [inline samples/sheets/quickstart.js]: ===

Comment (alexander-fenster) [reply samples/sheets/quickstart.js]: Looks like the CI is complaining about the same line :)

**Comment by grant** [comment]
@JustinBeckwith @alexander-fenster 

Thanks for the Sheets addition. I want to let you know that I'm releasing G Suite Node samples (quickstarts + snippets) for ~6 G Suite products in a week or so. I appreciate the effort to updating this one and putting it on GitHub, but it looks like this code is directly pulled from the quickstart with some touchups.

I've since added the dependency version to the [sheets quickstart](https://developers.google.com/sheets/api/quickstart/nodejs) so that it does break. It will be updated and linked to GitHub when the **`gsuitedevs/node-samples`** repo is published.

In order to not have duplicate code, and to match the style of the `samples/` that doesn't have quickstarts like this, would you consider adding a cross link to the G Suite samples instead of having 2 versions when the node-samples repo is published? It seems like adding this quickstart is trying to patch the root problem.

**Comment by JustinBeckwith** [comment]
>  I want to let you know that I'm releasing G Suite Node samples (quickstarts + snippets) for ~6 G Suite products in a week or so.
Yes!  This is amazing!  

> it looks like this code is directly pulled from the quickstart with some touchups.
That's because I submitted the CL to update the quickstart first 🤣 I wanted to get that updated quickly (people were getting led astray) while we worked on getting real samples checked in and tested.  

> I've since added the dependency version to the sheets quickstart so that it does break.
The other approach we could take here is to instruct users to install at head, but keep all the samples in this repo so they're always tested.  I'm a humble PM volunteering time here, and the tests in this lib far from cover all scenarios.  I want to make sure our samples are tested here so I know that I'm not breaking anyone.  

> In order to not have duplicate code, and to match the style of the samples/ that doesn't have quickstarts like this, would you consider adding a cross link to the G Suite samples instead of having 2 versions when the node-samples repo is published? It seems like adding this quickstart is trying to patch the root problem.
I would prefer to move all our node-samples for this library over here, for the reasons stated above.  As we do make breaking changes, we can make sure we're fixing the docs in the same PR that introduces the problem.  Also, it's always nice to have more node.js folks to look at PRs in these repos :)

This appears to have a lot of moving parts.  If you'd like to chat on Monday, happy to hop on a GVC!

**Comment by grant** [comment]
Thanks for volunteering time. I'm sure some sample overlap is fine.

> That's because I submitted the CL to update the quickstart first 🤣 I wanted to get that updated quickly (people were getting led astray) while we worked on getting real samples checked in and tested.

The best way to update the quickstart would be to send a message to the DevSite author saying that the docs sample is broken. Copying this one sample into another repo is not a great fix. There were 5 other product samples that I updated as well that weren't reported. Installing at HEAD is not a reliable solution for every language/product sample.

---

Some more background:

The G Suite samples are planned to be in a similar structure to GCP's samples:

https://github.com/GoogleCloudPlatform/nodejs-docs-samples

There are ~10 languages, so any solution of adding samples to client repos would have to be consistent will all clients:
https://github.com/google?utf8=%E2%9C%93&q=google-api-&type=&language=

There are complications (like no python client, multiple sample repos) that makes adding samples to client repos difficult. It's really a more complicated discussion that was talked about with OSPO/github-requests@ a few months ago.

If we want to start a discussion about consolidating all Google Node samples in this repo, including [GCP](https://github.com/GoogleCloudPlatform/nodejs-docs-samples), [this client library](https://github.com/google/google-api-nodejs-client), G Suite, [googlemaps](https://github.com/googlemaps/js-samples), etc. then we should start a conversation with Max/OSPO and all sample repo maintainers.

The main point I'd like to get across is that copying one G Suite sample from DevSite so there are **two versions** is not a great solution to fixing this problem. Simply ask the Google author for the docs to be fixed.

**Comment by JustinBeckwith** [comment]
I'm in total agreement that having two versions of these samples is a bad idea.  I didn't know y'all were writing and maintaining these samples - this is great!  

Now on the repo where the samples live - this is probably a discussion that needs to include @jonparrott @jmdobry and @ace-n.  Folks - what's the future of the `nodejs-docs-samples` repo?  I was under the understanding that we're actively trying to pull things out of there, and move samples closer to the repo with the actual implementations.  Is that accurate?

**Comment by grant** [comment]
Note: All G Suite docs and samples are maintained by G Suite DevRel.

I'll set up a meeting next week for coordinating these samples. Be wary that G Suite and `@google-cloud` are a bit different, and that any solution proposed should cover .NET, Android, Apps Script, Go, iOS, Java, JS (client-side), Node, PHP, Python and Ruby equally.

**Pull request merged**